src/main/java/frc/robot/subsystems/BallPath/Shooter/PIDShooterTrackingImpl.java

```diff
@@ -0,0 +1,363 @@
+        SmartDashboard.putNumber("Turret Encoder Reading Position", turretEncoderReadingPosition);
+        SmartDashboard.putNumber("Turret Encoder Reading Velocity", turretEncoderReadingVelocity);
+        SmartDashboard.putNumber("Turret Hood Encoder reading Position", turretHoodPosition);
+        SmartDashboard.putNumber("Turret Hood Encoder Reading Velocity", turretHoodVelocity);
+
+        switch (this.requestedPosition) {
+            case FENDER:
+                aim = false;
+                setPointShooterPID = 5_500;
+                setPointHood = 100_000;
+                break;
+            case GENERAL:
+                aim = true;
+                setPointHood = getSetpointHood(totalDistance);
+                setPointShooterPID = getSetpointWheel(totalDistance);
+                shoot = true;
+                break;
+            case NONE:
+                shoot = false;
+                setPointShooterPID = 2000;
+                aim = true;
+            case TEST:
+            default:
+                break;
+        }
+        // Getting setpoints and target position
+
+
+        // hood setpoint
+        if(shoot){
+            if (setPointHood == Double.NEGATIVE_INFINITY) {
+                hoodMotor.set(ControlMode.PercentOutput, 0);
+                hoodReady = false;
+            } else if(turretHoodPosition >= setPointHood - hoodBuffer && turretHoodPosition <= setPointHood + hoodBuffer){
+                hoodMotor.set(ControlMode.PercentOutput, 0);
+                hoodReady = true;
+            }else if(turretHoodPosition <= setPointHood - hoodBuffer){
+                hoodMotor.set(ControlMode.PercentOutput, hoodSpeed);
+                hoodReady = false;
+            }else if (turretHoodPosition >= setPointHood + hoodBuffer){
+                hoodMotor.set(ControlMode.PercentOutput, -hoodSpeed);
+                hoodReady = false;
+            }
+        }else{
+            hoodReady = false;
+        }
+
+        // turret movement
+        // if the user does not want to do a fender shot
+        if(aim){
+            // only get the distance if we are aiming
+            tx = table.getEntry("tx");
+            ty = table.getEntry("ty");
+            ta = table.getEntry("ta");
+            tv = table.getEntry("tv");
+            x = tx.getDouble(0.0);
+            y = ty.getDouble(0.0);
+            a = ta.getDouble(0.0);
+            canSeeTarget = tv.getDouble(0.0);
+            totalAngle = a1+y;
+            totalAngleRadians = Math.toRadians(totalAngle);
+            rs = Math.tan(totalAngleRadians);
+            totalDistance = heightDif / rs;
+            
+            if(turretEncoderReadingPosition > leftLimit && turretEncoderReadingPosition < rightLimit && !flipLeft && !flipRight){
+                if(canSeeTarget==1.0){
+                    // if we are in the encoder limits, can see the target, and do not want to "flip"
+                    if(x < 1 && x > -1){
```

Comment:
What does the `1` mean here for the `x` value? Is that the x-distance of the vision target from the center of the frame? If so, what are the units, and how did you arrive at `1` being the correct toleranced value?

Reply:
And in any case, please extract the `1` out to a named constant so that if you do need to update that, it can be done by changing one line and not having to hunt each reference. It also makes it more readable, which is arguably even more important.